### DISCUSSION TITLE
Unable to build documentation locally for pytorch

### DISCUSSION THREAD

**Discussion by davidradl** [pypa/setuptools#3968] [Q&A]
I am working with pytorch and am following the [documentation](https://github.com/pytorch/pytorch#from-source)  to build the documentation locally. I am pursuing this in issue https://github.com/pytorch/pytorch/issues/104113#issuecomment-1614787310

We run 
`python setup.py develop`
The processing sets up egg-links and egg-info files in the miniconda3 site_packages. 
The is a second package called torchgen in the top_level.txt.

When i run the make html to build the documentation it picks up files from miniconda3 rather than my local folder. my local folder in the egg link is not first in the path. Any advice as to how to get my local folder to be first in the path so it picks up the local code only. This is an issue as there is a mismatch between the miniconda3 and the local code - so there is a failure. 

I am thinking that if the egg links were correct as setup by the setup tools - then my  local code would be picked up.

Any advice would be greatly appreciated.

**Comment by webknjaz** [comment]
Calling `setup.py` directly is deprecated and has been actively discouraged for many years. I don't know what pytorch is doing with it but it's unsupported and ultimately, they are responsible for however they misuse setuptools. Hence, it's best to ask them for instructions. Also, Conda is a non-standard ecosystem that the wider community doesn't really interact with and setuptools is not tested under that runtime.

**Comment by sathesh4** [comment]
To resolve the issue of your local folder not being picked up first in the Python path when building the documentation for PyTorch, you can try the following steps:

1. Remove the previously installed package:
Since you have already run `python setup.py develop`, you need to undo that step first. To do this, go to the root directory of the PyTorch project and run:

```
python setup.py develop --uninstall
```

This will remove the package installed with `develop` mode.

2. Install the package in editable mode again:
Navigate to the root directory of your PyTorch project and run:

```
pip install -e .
```

This will install the package in editable mode, which will create a link to your local code, ensuring that your local folder is picked up first in the Python path.

3. Rebuild the documentation:
After reinstalling the package in editable mode, try running the command to build the documentation again:

```
make html
```

With the package installed in editable mode, it should now pick up your local code first, and any changes you make to the code will be reflected in the documentation.

By following these steps, you should be able to ensure that your local folder takes precedence in the Python path, and the documentation build process should use the local code instead of the one installed in the miniconda3 site-packages, avoiding any potential mismatches and failures.

**Reply by davidradl** [reply]
@sathesh4 thank you very much for your response. I have managed to get this to work, but not reliably. I am not sure what state the folders are in that does not make allow the -e to work; but it has worked for me.
